src/middlewares/roleAuthorization.js

```diff
@@ -1,13 +1,26 @@
 import Responses from '../utils/response';
 
-const roleAuthorization = {
-  checkIsSuperAdministrator(req, res, next) {
-    const { role } = res.locals.user;
-    if (role === 'super_administrator') {
-      next();
-    } else {
-      Responses.handleError(403, 'insufficient privileges', res);
-    }
+
+const checkIsSuperAdministrator = (req, res, next) => {
```

Comment:
I think these can be written in one function and pass in parameter an array of allowed roles:
REF: https://stackoverflow.com/questions/50989393/pass-parameter-to-middleware-function-in-expressjs

Reply:
I tried something like that but I failed because middleware was passing empty `req` and `res` to the next middleware. maybe we can work on that together because it is effective that creating a function for each user.